addons/l10n_in_pos/static/src/overrides/models/order.js

```diff
@@ -9,6 +9,95 @@ patch(Order.prototype, {
+            const taxes_ids = line.tax_ids || line.product.taxes_id;
+            if (hsnCode && taxes_ids) {
+                const product_taxes = line.pos.get_taxes_after_fp(taxes_ids, fiscalPosition);
+                const price_unit = line.get_unit_price();
+                const quantity = line.get_quantity()
+                const all_taxes = line.compute_all(product_taxes, price_unit, quantity, line.pos.currency.rounding);
+                //It is require to pre-compute GST Rate as we may have same hsn with different gst rates
+                const gstRate = this._computeL10nInGSTRate(all_taxes.taxes)
+                const groupKey = ''.concat(hsnCode, '-', gstRate)
+                if (!hsnSummary.hasOwnProperty(groupKey)){
+                    hsnSummary[groupKey] = {
+                        l10n_in_hsn_code: hsnCode,
+                        gst_rate: gstRate,
+                        SGST: {rate: 0, amount: 0},
+                        CGST: {rate: 0, amount: 0},
+                        IGST: {rate: 0, amount: 0},
+                        CESS: {amount: 0},
+                    }
+                }
+                const l10nInTaxKeyByTagRef = this._getl10nInTaxKeyByTagRef()
+                all_taxes.taxes.forEach((tax) => {
+                    tax.repartition_line_ids.forEach((line) => {
+                        line.tag_ids.forEach((tag) => {
+                            let tagRef = accountTagByXmlRefId[tag.id]
+                            let factor_percent = line.factor_percent
+                            const taxKey = l10nInTaxKeyByTagRef[tagRef]
+                            switch (taxKey) {
+                                case 'CGST':
+                                case 'SGST':
+                                case 'IGST':
+                                    hsnSummary[groupKey][taxKey]['amount'] += factor_percent * tax.amount / 100
+                                    hsnSummary[groupKey][taxKey]['rate'] = factor_percent * tax.tax_rate / 100
+                                    break;
+                                case 'CESS':
+                                    hsnSummary[groupKey][taxKey]['amount'] += factor_percent * tax.amount / 100
+                                    break;
+                            }
+                        })
+                    })
+                });
+            }
+        });
+        const hsnSummaryLines = Object.keys(hsnSummary)
+        const currency = this.pos.currency
+        hsnSummaryLines.forEach(line => {
+            const line_value = hsnSummary[line]
+            line_value.CGST.amount = this.env.utils.formatCurrency(line_value.CGST.amount, currency)
+            line_value.SGST.amount = this.env.utils.formatCurrency(line_value.SGST.amount, currency)
+            line_value.IGST.amount = this.env.utils.formatCurrency(line_value.IGST.amount, currency)
+            line_value.CESS.amount = this.env.utils.formatCurrency(line_value.CESS.amount, currency)
```

Comment:
you should also add lines here that formatFloat the `rate`. I believe you don't want the `rate` to be displayed with many decimals.

Reply:
Sorry, no need for formatFloat. I just noticed that you are not adding the rates.